### DISCUSSION TITLE
Solved: Command not saved

### DISCUSSION THREAD

**Discussion by chrisgrieser** [Taitava/obsidian-shellcommands#149] [Debugging and testing]
See this video documenting the problem. I literally cannot save a single command :(

https://user-images.githubusercontent.com/73286100/153489652-5eae3627-ff4f-4092-bebb-c212a1c032c9.mov

**Comment by Taitava** [comment]
Hi, thank you for your report! Really good that you made a video! 🙂

Can you plese do these steps:
1. Edit `.obsidian/plugins/obsidian-shellcommands/data.json` in a text editor and change `debug` to `true`. Then restart Obsidian.
2. Press `Cmd` (or `Ctrl`) + `Shift` + `I` to open up Obsidian Developer Console.
3. Record a new video where you redo the test with the developer console open. Some error messages might appear there giving me some kind of clue what goes wrong.
4. You can then change `debug` back to `false`.

(Step 1 is optional and it gives just a bit more output in the developer console, telling what the plugin is doing. A text that appears just before a possible error message can give me furher information about in which part of the plugin's code the problem might occur.)

Thanks! 🙂

**Reply by chrisgrieser** [reply]
well, there isn't much to record in a video – it's just one error message.

```shell
# entered into the field
open -a "Hyper" "{{folder_name}}"
```
```text
Uncaught TypeError: Cannot read property 'type' of undefined
    at Variable_FolderName.setArgument (eval at <anonymous> (app.js:1), <anonymous>:1094:68)
    at parseShellCommandVariables (eval at <anonymous> (app.js:1), <anonymous>:2462:30)
    at getShellCommandPreview (eval at <anonymous> (app.js:1), <anonymous>:3178:32)
    at on_change (eval at <anonymous> (app.js:1), <anonymous>:3145:47)
    at e.onChanged (app.js:1)
```

according to the log, there seems to be a problem with the `{{folder_name}}`, so I tried a different variable, also no luck. But trying something entirely without variables like `echo "foobar"` does work, so I assume it's something about variables that's causing the issue here?

**Reply by Taitava** [reply]
Thanks. I think it looks like the same problem as in #137 . When _Dynamic highlights_ and _Shell commands_ are both installed, SC stops working. I don't know which one of the plugins causes the problem. [DH has an issue created for it](https://github.com/nothingislost/obsidian-dynamic-highlights/issues/25), but it looks like the inspection is still in an early stage.

It's hard from both plugin developers' point of view: I haven't used DH, and @nothingislost hasn't used SC, and digging into an unknown plugin's source code can take some effort.

Well, now that I took a really quick glance over DH's source code, I noticed it's not too big, and I might take a better look at it later.

**Reply by chrisgrieser** [reply]
well that is indeed weird for two plugins to have such a specific interaction-bug.

My best guess is that it has something to do with [this bug I noticed with DH](https://github.com/nothingislost/obsidian-dynamic-highlights/issues/30), where the `esc` key doesn't work in the settings tab anymore?

**Reply by Taitava** [reply]
Yes, it's weird. To be honest, I don't know JavaScript imports very well, nor what kind of isolation there is between different Obsidian plugins. So I'm not aware of all the possible ways plugins can interfere with each other.

**Reply by Taitava** [reply]
And that esc bug seems interesting, too.

**Reply by chrisgrieser** [reply]
Just tried it, deactivating Dynamic Highlights allows me to enter the command needed.

**Reply by Taitava** [reply]
Good to know that the relation to DH is confirmed also in this case. There's still a long way till finding a solution, but let's hope for the best.

For me, the easiest thing to begin with is to checkout earlier versions of SC and see if there happpens to be some early version that will not get blocked by DH. After that, I will start looking into DH's source code.

**Reply by nothingislost** [reply]
See here for more detail: https://github.com/nothingislost/obsidian-dynamic-highlights/issues/25#issuecomment-1038563990

edit: Your plugin is great, btw! Really excellent job on the settings UI/UX

**Reply by Taitava** [reply]
Thank you for the nice feedback! ❤️ I'm not really a UX/UI designer, so it's nice to hear that I have happened to do it well. 🙂

**Comment by chrisgrieser** [comment]
with the latest update of Dynamic Highlights, the issue seems to be resolved for me.
https://github.com/nothingislost/obsidian-dynamic-highlights/issues/25#issuecomment-1039254726
